db/migrate/20150525081931_remove_duplicate_tokens.rb

```diff
@@ -0,0 +1,27 @@
+class RemoveDuplicateTokens < ActiveRecord::Migration
+  def up
+    # only managed hosts should have tokens
+    Token.where(:id => Token.joins("left outer join hosts on hosts.id = tokens.host_id").where("hosts.type != 'Host::Managed'").pluck(:id)).delete_all
```

Comment:
`Token.joins("left outer join hosts on hosts.id = tokens.host_id").where("hosts.type != 'Host::Managed'").delete_all` resulted in an error:
```
ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  missing FROM-clause entry for table "hosts"
LINE 1: DELETE FROM "tokens" WHERE (hosts.type != 'Host::Managed')
```
ideas on how to delete these records without using the tokens table twice?